WcaOnRails/app/webpacker/components/Panel/Wrt/EditPerson.jsx

```diff
@@ -0,0 +1,183 @@
+  It will modify the Persons table accordingly and the Results table if the person's name is different.
+  This should be used to fix mistakes in the database.
+  </p>
+
+  <p>
+  Choose 'Update' if the person's name or country has been changed.
+  It will add a new entry in the Persons table and make it the current information for that person (subId=1)
+  but it will not modify the Results table so previous results keep the old name.
+  </p>
+`;
+
+const dateFormat = 'YYYY-MM-DD';
+
+const genderOptions = _.map(genders.byId, (gender) => ({
+  key: gender.id,
+  text: gender.name,
+  value: gender.id,
+}));
+
+const countryChangeWarning = `The change you made may have affected national and continental records, be sure to run <a href=${adminCheckRecordsUrl}>check_regional_record_markers</a>.`;
+
+function EditPerson({ countryList }) {
+  const [person, setPerson] = React.useState();
+  const [editedUserDetails, setEditedUserDetails] = React.useState();
+  const [originalUserDetails, setOriginalUserDetails] = React.useState();
+  const [incorrectClaimCount, setIncorrectClaimCount] = useState(0);
+  const { save, saving } = useSaveAction();
+  const [response, setResponse] = useState();
+
+  React.useEffect(() => {
+    if (person) {
+      const userDetails = {
+        wcaId: person.item.wca_id,
+        name: person.item.name,
+        representing: person.item.country_iso2,
+        gender: person.item.gender,
+        dob: person.item.dob,
+      };
+      setOriginalUserDetails(userDetails);
+      setEditedUserDetails(userDetails);
+      setIncorrectClaimCount(person.item.incorrect_wca_id_claim_count);
+      setResponse(null);
+    } else {
+      setOriginalUserDetails(null);
+      setEditedUserDetails(null);
+      setIncorrectClaimCount(0);
+    }
+  }, [person]);
+
+  const handleFormChange = (e, { name: formName, value }) => {
+    setEditedUserDetails((prev) => ({ ...prev, [formName]: value }));
+  };
+
+  const editPerson = (method) => {
+    save(adminUpdatePersonUrl, {
+      person: editedUserDetails,
+      method,
+    }, () => {
+      setResponse({
+        success: true,
+        message: `Success. ${originalUserDetails.representing !== editedUserDetails.representing
+          ? countryChangeWarning : ''}
+          </p>`,
+      });
+      setPerson(null);
+    }, {}, (error) => {
+      setResponse({
+        success: false,
+        message: `${error}`,
+      });
+    });
+  };
+
+  if (saving) return <Loading />;
+
+  return (
+    <>
+      {/* eslint-disable-next-line react/no-danger */}
+      <div dangerouslySetInnerHTML={{ __html: description }} />
+      {response != null && (
+        <Message
+          success={response.success}
+          error={!response.success}
+          content={response.message}
+        >
+          <Message.Content>
+            {/* eslint-disable-next-line react/no-danger */}
+            <div dangerouslySetInnerHTML={{ __html: response.message }} />
```

Comment:
Why not just `<Message.Content>{response.message}</Message.Content>`?

Reply:
Because the message can contain hyperlink (like the check records link).

Reply:
Then make the message return some error code and the raw link URL separately, and load the i18n'd version here (which will also allow you to inject the link by using `I18nTranslate` React component).

I have very strong reservations about allowing more `dangerouslySetInnerHTML` in our codebase.